solution/0100-0199/0100.Same Tree/Solution.php

```diff
@@ -0,0 +1,32 @@
+/**
+ * Definition for a binary tree node.
+ * class TreeNode {
+ *     public $val = null;
+ *     public $left = null;
+ *     public $right = null;
+ *     function __construct($val = 0, $left = null, $right = null) {
+ *         $this->val = $val;
+ *         $this->left = $left;
+ *         $this->right = $right;
+ *     }
+ * }
+ */
+class Solution {
+    /**
+     * @param TreeNode $p
+     * @param TreeNode $q
+     * @return Boolean
+     */
+    function isSameTree($p, $q) {
+        if ($p == Null && $q == Null) {
+            return true;
+        }
+        if ($p == Null || $q == Null) {
+            return false;
+        }
+        if ($p -> val != $q -> val) {
+            return false;
+        }
+        return $this -> isSameTree($p -> left, $q -> left) && $this -> isSameTree($p -> right, $q -> right);
```

Comment:
虽然支持 `$this -> propsName` 这种写法，但推荐使用 `$this->propsName` 形式（箭头左右不加空白，可参考该文件 8~10 行注释）。

```suggestion 
        if ($p->val != $q->val) {
            return false;
        }
        return $this->isSameTree($p->left, $q->left) && $this->isSameTree($p->right, $q->right);
```